src/audio/module_adapter/module/volume/volume.c

```diff
@@ -1300,7 +1300,13 @@ static int volume_prepare(struct processing_module *mod)
 	 * for entire topology specified time.
 	 */
 	cd->ramp_finished = false;
+
 	cd->channels = sink_c->stream.channels;
+	if (cd->channels > SOF_IPC_MAX_CHANNELS) {
```

Comment:
if this is an array, won't we need to check >= here ?

Reply:
@lgirdwood  The code will operate on the arrays with "for (i = 0; i < cd->channels; i++)" loops so it's ok to have ch->channels==SOF_IPC_MAX_CHANNELS.